design/operator-architecture.asciidoc

```diff
@@ -0,0 +1,154 @@
+# Keycloak Operator High Level Architecture
+
+* **Status**: Draft #1
+* **JIRA**: https://issues.redhat.com/browse/KEYCLOAK-10319[KEYCLOAK-10319]
+* **Source of the images**: https://docs.google.com/presentation/d/13N5ClXXXcxKjXgor72rUdtMBN07_GneUkhTTwBsy_tE/edit?usp=sharing[Google Slides]
+
+This document shows a big picture how all the components and layers of Keycloak Operator work together. The blueprint presented here shall be used a guideline for designing features as well as integrating community Pull Requests.
+
+## Manifesto
+
+We envision Keycloak Operator as:
+
+* an opinionated way of installing and maintaining Keycloak installation on both Kubernetes and OpenShift.
+* a way to deliver a high level usability rather than low-level toggles
+* an example of a recommended way of installing, configuring, managing custom extension and themes.
+* the thinnest possible layer, with the least possible logic, on top of Keycloak Container and Keycloak Server.
```

Comment:
I think I understand what you mean, but this could also be interpreted as just dumping the exported config into map[string]string.

Do you envision being able to set up and configure keycloak just by reading the documentation/CRD and creating a CRD, instead doing the setup in the UI and exporting the configuration (including the "defaults" that are included in the export)?

Reply:
Our CRDs reflect Keycloak resources. A good example is `RealmRepresentation` JSON payload and `KeycloakRealm` CRD. They are very similar and, in the long-term, should be the same.

So it is perfectly doable to configure Keycloak purely through CRDs (without touching the UI).